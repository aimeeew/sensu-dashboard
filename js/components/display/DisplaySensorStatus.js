import React from 'react'

const DisplaySensorStatus = ({data}) => {

    //needs changing to be async
    if (!data) {
        return <div/>
    }

    function getService(name)
    {
        return (name.split(".", 2)).join(".");
    }

    function getName(datum)
    {
        return datum.check.name.replace("Mayden.Sensors.", "");
    }

    function getStatusColour(datum)
    {
        return datum.check.status === 0 ? "statusItemGreen" : "statusItemRed";
    }

    function getDateTime(datum)
    {
        let date = new Date(datum.check.executed * 1000);
        let dateString = date.toLocaleDateString();
        let time = date.getHours() + ':' + ("0" + date.getMinutes()).substr(-2) + ':' + ("0" + date.getSeconds()).substr(-2);

        return time + " " + dateString;
    }

    function getMessage(datum)
    {
        return getStatusColour(datum) === "statusItemRed" ?  datum.check.output : "";
    }

    return (
        <div>
            <div className="grid-container">
                <div className="serviceHeading">Service</div>
                <div className="nameHeading">Name</div>
                <div className="statusHeading">Status</div>
                <div className="statusCodeHeading">(Code)</div>
                <div className="timeHeading">Time</div>
                <div className="messageHeading">Message</div>
            </div>

            {
                data.map((datum, index) => {

                    return (
                        <div className="grid-container" key={index}>
                            <div className="serviceItem">{getService(getName(datum))}</div>
                            <div className="nameItem">{getName(datum).replace(getService(getName(datum)) + ".", "")}</div>
                            <div className ={getStatusColour(datum) + "Dot"}/>
                            <div className="statusCodeItem">({datum.check.status})</div>
                            <div className="timeItem">{getDateTime(datum)}</div>
                            <div className="messageItem">{getMessage(datum)}</div>
                        </div>
                    )
                })
            }
        </div>
    )
}
export default DisplaySensorStatus;
