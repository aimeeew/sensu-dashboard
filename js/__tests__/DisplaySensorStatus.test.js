import React from 'react';
import renderer from "react-test-renderer";
import DisplaySensorStatus from "../components/display/DisplaySensorStatus";

const data =
    [{
        "client": "sensu-client-ivnode14",
        "check": {
            "type": "metric",
            "command": "php /var/www/html/sensor-manager/SensorManager.php run Iaptus Availabilities\\\\Metrics\\\\AvailabilityCreatedCount",
            "subscribers": [
                "iaptus"
            ],
            "interval": 450,
            "handlers": [
                "relay"
            ],
            "name": "Mayden.Sensors.Iaptus.Availabilities.Metrics.AvailabilityCreatedCount",
            "issued": 1563781786,
            "executed": 1563781813,
            "duration": 17.94,
            "output": "sensu.iaptus.bracknell.iaptusvnode14.availability_count.12_gloucester 1 1563781830\n...",
            "status": 0
        }
    },
        {
            "client": "sensu-client-cron01.iaptus.ukc",
            "check": {
                "command": "php /var/www/html/sensor-manager/SensorManager.php run Iaptus Sms\\\\Checks\\\\SendingOnTime",
                "subscribers": [
                    "iaptus"
                ],
                "interval": 1800,
                "handlers": [
                    "mailer",
                    "slack"
                ],
                "name": "Mayden.Sensors.Iaptus.Sms.Checks.SendingOnTime",
                "issued": 1562772496,
                "executed": 1562772513,
                "duration": 13.33,
                "output": "{\"errors\":[{\"value\":\"1\"},{\"value\":\"16\"}]}\n",
                "status": 2,
                "type": "standard"
            }
        },
    ]

describe('Display Component', () => {
    it('matches the snapshot', () => {
        const tree = renderer
            .create(<DisplaySensorStatus data={data}/>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})