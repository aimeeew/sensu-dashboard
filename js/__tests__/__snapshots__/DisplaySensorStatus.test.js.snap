// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Component matches the snapshot 1`] = `
<div>
  <div
    className="grid-container"
  >
    <div
      className="serviceHeading"
    >
      Service
    </div>
    <div
      className="nameHeading"
    >
      Name
    </div>
    <div
      className="statusHeading"
    >
      Status
    </div>
    <div
      className="statusCodeHeading"
    >
      (Code)
    </div>
    <div
      className="timeHeading"
    >
      Time
    </div>
    <div
      className="messageHeading"
    >
      Message
    </div>
  </div>
  <div
    className="grid-container"
  >
    <div
      className="serviceItem"
    >
      Iaptus.Availabilities
    </div>
    <div
      className="nameItem"
    >
      Metrics.AvailabilityCreatedCount
    </div>
    <div
      className="statusItemGreenDot"
    />
    <div
      className="statusCodeItem"
    >
      (
      0
      )
    </div>
    <div
      className="timeItem"
    >
      8:50:13 7/22/2019
    </div>
    <div
      className="messageItem"
    >
      
    </div>
  </div>
  <div
    className="grid-container"
  >
    <div
      className="serviceItem"
    >
      Iaptus.Sms
    </div>
    <div
      className="nameItem"
    >
      Checks.SendingOnTime
    </div>
    <div
      className="statusItemRedDot"
    />
    <div
      className="statusCodeItem"
    >
      (
      2
      )
    </div>
    <div
      className="timeItem"
    >
      16:28:33 7/10/2019
    </div>
    <div
      className="messageItem"
    >
      {"errors":[{"value":"1"},{"value":"16"}]}

    </div>
  </div>
</div>
`;
